lib/signatures.ts

```diff
@@ -3,35 +3,41 @@ import { SignatureHashAlgorithm, SignatureResult } from './types';
 import { getDubHashSignature } from './dubhash';
 import { getUhashSignature } from './uhash';
 import pMap = require('p-map');
+import { join } from 'path';
 
 const { readFile } = promises;
 
 export async function getSignaturesByAlgorithm(
+  baseDir: string,
   filePaths: string[],
   hashType: SignatureHashAlgorithm = 'dubhash',
 ): Promise<(SignatureResult | null)[]> {
   if (hashType !== 'dubhash' && hashType !== 'uhash') {
     throw new Error(`Unsupported hashType ${hashType}`);
   }
-  let signatureMapperToBeUsed = getDubHashSignatureMapper;
+  let signatureMapperToBeUsed = getDubHashSignatureMapper(baseDir);
   if (hashType === 'uhash') {
-    signatureMapperToBeUsed = getUHashSignatureMapper;
+    signatureMapperToBeUsed = getUHashSignatureMapper(baseDir);
   }
-  return await pMap(filePaths, signatureMapperToBeUsed, { concurrency: 20 });
+  return await pMap(filePaths, signatureMapperToBeUsed, {
+    concurrency: 20,
+  });
 }
 
-async function getDubHashSignatureMapper(
-  filePath: string,
-): Promise<SignatureResult | null> {
-  const fileContents = await readFile(filePath);
-  if (fileContents.length === 0) return null;
-  return getDubHashSignature(filePath, fileContents);
-}
+const getDubHashSignatureMapper = (baseDir: string) =>
```

Comment:
would suggest to keep the function as `function` rather than `const` as it have an effect on readability and error stacktrace